spec/factories/subjects/secondary_subjects.rb

```diff
@@ -0,0 +1,197 @@
+FactoryBot.define do
+  subjects = [
+      # Modern Lanuages has been removed as it would cause validation errors if it
+      # isn't paired with a valid modern language subject use a trait if you want to create it
+      { subject_name: "Art and design", subject_code: "W1" },
+      { subject_name: "Science", subject_code:  "F0" },
+      { subject_name: "Biology", subject_code:  "C1" },
+      { subject_name: "Business studies", subject_code: "08" },
+      { subject_name: "Chemistry", subject_code: "F1" },
+      { subject_name: "Citizenship", subject_code:  "09" },
+      { subject_name: "Classics", subject_code: "Q8" },
+      { subject_name: "Communication and media studies", subject_code: "P3" },
+      { subject_name: "Computing", subject_code: "11" },
+      { subject_name: "Dance", subject_code: "12" },
+      { subject_name: "Design and technology", subject_code: "DT" },
+      { subject_name: "Drama", subject_code: "13" },
+      { subject_name: "Economics", subject_code: "L1" },
+      { subject_name: "English", subject_code: "Q3" },
+      { subject_name: "Geography", subject_code: "F8" },
+      { subject_name: "Health and social care", subject_code: "L5" },
+      { subject_name: "History", subject_code:  "V1" },
+      { subject_name: "Mathematics", subject_code: "G1" },
+      { subject_name: "Music", subject_code: "W3" },
+      { subject_name: "Philosophy", subject_code: "P1" },
+      { subject_name: "Physical education", subject_code: "C6" },
+      { subject_name: "Physics", subject_code: "F3" },
+      { subject_name: "Psychology", subject_code: "C8" },
+      { subject_name: "Religious education", subject_code: "V6" },
+      { subject_name: "Social sciences", subject_code: "14" },
+    ]
```

Comment:
As with primary subjects, this can be better expressed as a Hash of subject name to code, imo. I made a quick macro to convert from what you have here to what I'm suggesting, so in case you want to use it I'll include the converted version of the above for your convenience. :)

```ruby
  subjects = {
      # Modern Lanuages has been removed as it would cause validation errors if it
      # isn't paired with a valid modern language subject use a trait if you want to create it
     "Art and design"                  => "W1",
     "Science"                         => "F0",
     "Biology"                         => "C1",
     "Business studies"                => "08",
     "Chemistry"                       => "F1",
     "Citizenship"                     => "09",
     "Classics"                        => "Q8",
     "Communication and media studies" => "P3",
     "Computing"                       => "11",
     "Dance"                           => "12",
     "Design and technology"           => "DT",
     "Drama"                           => "13",
     "Economics"                       => "L1",
     "English"                         => "Q3",
     "Geography"                       => "F8",
     "Health and social care"          => "L5",
     "History"                         => "V1",
     "Mathematics"                     => "G1",
     "Music"                           => "W3",
     "Philosophy"                      => "P1",
     "Physical education"              => "C6",
     "Physics"                         => "F3",
     "Psychology"                      => "C8",
     "Religious education"             => "V6",
     "Social sciences"                 => "14",
  }
```